$tmpl produce wrong code inside anonymous function

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 3a65ccd
**OS:**  Windows 10

**What did you do?**
```v
fn main() {
  anon := fn (data string) string {
    return $tmpl('1.txt') // println("@data")
  }

  println(anon('hello'))
}
```

**What did you expect to see?**
Working program
 
**What did you see instead?**
Wrong C code.

```c
string anon_98_18_18(string data) {
	strings__Builder sb = strings__new_builder(30);
	string header = _SLIT(" ");
	{ string _ = header; }
	;
	string footer = _SLIT(" ");
	{ string _ = footer; }
	;
	strings__Builder_write(&sb, _STR("println(\"%.*s\000\")\n", 2, data));
	string _tmpl_res_main__main = strings__Builder_str(&sb);
	return _tmpl_res_anon_98_18_18;
  ;
}
```
tmpl_bug.tmp.c(1560): error C2065: '_tmpl_res_anon_98_18_18': undeclared identifier

--- a/vlib/v/parser/fn.v
+++ b/vlib/v/parser/fn.v
@@ -551,6 +551,14 @@ fn (mut p Parser) anon_fn() ast.AnonFn {
 			p.tok.position())
 	}
 	mut label_names := []string{}
+	mut func := table.Fn{
+		params: args
+		is_variadic: is_variadic
+		return_type: return_type
+	}
+	name := 'anon_fn_${p.table.fn_type_signature(func)}_$p.tok.pos'
+	keep_fn_name := p.cur_fn_name
+	p.cur_fn_name = name
 	if p.tok.kind == .lcbr {
 		tmp := p.label_names
 		p.label_names = []
@@ -558,13 +566,8 @@ fn (mut p Parser) anon_fn() ast.AnonFn {
 		label_names = p.label_names
 		p.label_names = tmp
 	}
+	p.cur_fn_name = keep_fn_name
 	p.close_scope()
-	mut func := table.Fn{
-		params: args
-		is_variadic: is_variadic
-		return_type: return_type
-	}
-	name := 'anon_fn_${p.table.fn_type_signature(func)}_$p.tok.pos'
 	func.name = name
 	idx := p.table.find_or_register_fn_type(p.mod, func, true, false)
 	typ := table.new_type(idx)

--- a/vlib/v/tests/tmpl_test.v
+++ b/vlib/v/tests/tmpl_test.v
@@ -19,3 +19,23 @@ numbers: [1, 2, 3]
 
 3'
 }
+
+fn test_tmpl_in_anon_fn() {
+	anon := fn (name string, age int, numbers []int) string {
+		return $tmpl('tmpl/1.txt')
+	}
+
+	println(anon('Peter', 25, [1, 2, 3]))
+	assert anon('Peter', 25, [1, 2, 3]).trim_space() == 'name: Peter
+
+age: 25
+
+numbers: [1, 2, 3]
+
+
+1
+
+2
+
+3'
+}

